airbyte-ci/connectors/connector_ops/connector_ops/utils.py

```diff
@@ -325,6 +325,35 @@ def get_local_dependency_paths(self, with_test_dependencies: bool = True) -> Set
                 self.code_directory / "build.gradle", with_test_dependencies=with_test_dependencies
             )
         return sorted(list(set(dependencies_paths)))
+    
+
+@dataclass(frozen=True)
+class CDK:
+    """Utility class to gather metadata about a cdk."""
+
+    language: str
+
+    @property
+    def code_directory(self) -> Path:
+        return Path(f"./airbyte-cdk/{self.language}")
+
+
+    @property
+    def version(self) -> str:
+        if self.language == "java":
+            with open(self.code_directory / "resources/version.properties") as f:
+                for line in f:
+                    key, value = line.strip().split('=', 1)
+                    if key == 'version':
+                        return value
```

Comment:
nit: I think this can be implemented in a more elegant way with a ConfigParser:
https://docs.python.org/3/library/configparser.html